test/integ/test_cursor.py

```diff
@@ -1306,9 +1306,18 @@ def test_resultbatch_lazy_fetching(conn_cnx, result_format, patch_path):
             #  first fetchone call
             with mock.patch(patch_path, return_value=iter([(1,)])) as patched_download:
                 cur.execute(
-                    f"select seq4() from table(generator(rowcount => {rowcount}));"
+                    f"select seq4() as c1, randstr(1,random()) as c2 "
+                    f"from table(generator(rowcount => {rowcount}));"
                 )
                 result_batches = cur.get_result_batches()
+                batch_schemas = [batch.schema for batch in result_batches]
```

Comment:
I assert that the schema that the is available before the download, so that test verifies this case